core/dbt/config/profile.py

```diff
@@ -52,7 +52,7 @@ def read_profile(profiles_dir: str) -> Dict[str, Any]:
     contents = None
     if os.path.isfile(path):
         try:
-            contents = load_file_contents(path, strip=False)
+            contents = SA.adapter.load_file_contents(path, strip=False)  # type: ignore
```

Comment:
I think that profile reading should not go through the SA as strange as that sounds, or at least not go through the same load_file_contents method.

In a thin client world, the thin client will sync project file changes only -- the profile will need to come from somewhere else, and won't be stored alongside project files.